### PR TITLE
[Feature] new privilege framework: basic authentication

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Introduce a new privilege framework. 
To avoid huge PR which is hard to review and easy to conflict, I'm splitting into several small PR, each implementing an individual aspect of the new privilege framework. This is the first one.
This PR implements basic authentication and supports creating a user with MySQL plain password.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/authentication/UserProperty.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.authentication;
+
+import com.google.gson.annotations.SerializedName;
+
+public class UserProperty {
+    @SerializedName(value = "m")
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/authentication/UserProperty.java:8]: Set serializeName to "m" to save json size?

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/authentication/UserProperty.java:8]: Sí

**Comment by padmejin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 206 / 268 (76.87%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/execution/CreateUserExecutor.java|0|3|00.00%|[14, 15, 17]|
|:large_blue_circle:|com/starrocks/mysql/MysqlProto.java|1|10|10.00%|[60, 61, 63, 64, 65, 67, 68, 69, 71]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|1|2|50.00%|[139]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationProviderFactory.java|8|14|57.14%|[25, 26, 28, 29, 33, 34]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|6|10|60.00%|[888, 889, 890, 891]|
|:large_blue_circle:|com/starrocks/authentication/UserProperty.java|3|5|60.00%|[12, 13]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|17|25|68.00%|[1174, 1175, 1235, 1236, 1425, 1426, 1473, 1474]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzerV2.java|19|27|70.37%|[36, 40, 45, 59, 60, 61, 62, 66]|
|:large_blue_circle:|com/starrocks/authentication/UserAuthenticationInfo.java|26|36|72.22%|[56, 57, 58, 59, 65, 66, 77, 81, 97, 98]|
|:large_blue_circle:|com/starrocks/qe/ConnectScheduler.java|3|4|75.00%|[124]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|69|79|87.34%|[98, 99, 105, 129, 130, 131, 132, 154, 155, 159]|
|:large_blue_circle:|com/starrocks/persist/CreateUserInfo.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/PlainPasswordAuthenticationProvider.java|31|31|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationException.java|2|2|100.00%|[]|

**Review by gengjun-git — DISMISSED** [review]

**Review by waittttting — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 221 / 260 (85.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/execution/CreateUserExecutor.java|0|3|00.00%|[14, 15, 17]|
|:large_blue_circle:|com/starrocks/mysql/MysqlProto.java|1|10|10.00%|[60, 61, 63, 64, 65, 67, 68, 69, 71]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|1|2|50.00%|[139]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|6|10|60.00%|[888, 889, 890, 891]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|17|25|68.00%|[1174, 1175, 1235, 1236, 1425, 1426, 1473, 1474]|
|:large_blue_circle:|com/starrocks/qe/ConnectScheduler.java|3|4|75.00%|[124]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzerV2.java|21|27|77.78%|[45, 59, 60, 61, 62, 66]|
|:large_blue_circle:|com/starrocks/authentication/UserAuthenticationInfo.java|26|32|81.25%|[56, 57, 58, 59, 65, 66]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationManager.java|76|77|98.70%|[155]|
|:large_blue_circle:|com/starrocks/persist/CreateUserInfo.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/PlainPasswordAuthenticationProvider.java|31|31|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/journal/JournalEntity.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationProviderFactory.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/UserProperty.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationException.java|2|2|100.00%|[]|

**Comment by padmejin** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**